### PR TITLE
ES6 refactor: utils

### PR DESCRIPTION
**Pull request by felixfbecker**
See #6009

### CODE REVIEW

**Comment by mickhansen** [comment]
LGTM

File: lib/utils.js [hunk 1]
```diff
@@ -1,444 +1,497 @@
...
+
+function spliceStr(str, index, count, add) {
+  return str.slice(0, index) + add + str.slice(index + count);
+}
+exports.spliceStr = spliceStr;
+
+function camelize(str) {
+  return str.trim().replace(/[-_\s]+(.)?/g, (match, c) => c.toUpperCase());
+}
+exports.camelize = camelize;
```
Comment (felixfbecker) [inline lib/utils.js]: `_.camelCase`?

File: lib/utils.js [hunk 2]
```diff
@@ -1,444 +1,497 @@
...
+  }
+  return now;
+}
+exports.now = now;
+
+function tick(func) {
+  const tick = (global.hasOwnProperty('setImmediate') ? global.setImmediate : process.nextTick);
+  tick(func);
+}
+exports.tick = tick;
```
Comment (felixfbecker) [inline lib/utils.js]: Can I remove this since we are targeting Node 4?

Comment (janmeier) [inline lib/utils.js]: Yep

Comment (felixfbecker) [inline lib/utils.js]: It doesn't seem to be used anywhere anyway.

File: lib/utils.js (hunk 1 above)
Comment (janmeier) [inline lib/utils.js]: Or we could just use inflection https://github.com/dreamerslab/node.inflection#inflectioncamelize-str-low_first_letter-

Comment (felixfbecker) [inline lib/utils.js]: The only difference I see is that inflection only works on underscore => camelCase while lodash transforms from anything (kebab-case, PascalCase, CONSTANT_CASE) to camelCase. I already checked, `camelize` is used only in `camelizeIf`, so it is super easy to replace. I would vote for lodash's.

Comment (janmeier) [inline lib/utils.js]: 2nd'ed

**Comment by felixfbecker** [comment]
Alright, the inflection stuff can be done later. Data types refactor depends on this ^^

**Comment by felixfbecker** [comment]
Haha, seems like the old `camelize` didn't actually convert to camelCase, but kept the first capital letter:
```
2058      + expected - actual
2059
2060       [
2061         "id"
2062         "priority"
2063      -  "productId"
2064      -  "tagId"
2065      +  "ProductId"
2066      +  "TagId"
2067       ]
```

File: lib/utils.js [hunk 3]
```diff
@@ -1,444 +1,491 @@
...
+  s = s.replace(/(\/\*[\n\r\s\S]*?\*\/)/mg, '');
+
+  return s;
+}
+exports.removeCommentsFromFunctionString = removeCommentsFromFunctionString;
+
+function toDefaultValue(value) {
+  if (typeof value === 'function') {
+    const tmp = value();
+    if (tmp instanceof DataTypes.ABSTRACT) {
```
Comment (SimonSchick) [inline lib/utils.js]: You can omit the `else`s here as you return in every single block, less code and easier to read imo.

Comment (felixfbecker) [inline lib/utils.js]: can't we do that in a different PR? This is really only about ES6. I have other PRs that depend on this one.

Comment (janmeier) [inline lib/utils.js]: Yep, lets hold of on style for now - You are welcome to provide a PR @SimonSchick
